### PR TITLE
br: skip loading stats into memory if set `--load-stats` to false

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: ref #50568

Problem Summary:
there is no necessary to load stats data into memory if set `--load-stats` to false.
### What changed and how does it work?
skip loading stats into memory if set `--load-stats` to false
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client.go
Comment (3pointer) [inline br/pkg/restore/client.go:1909]: Will row_count in stats_meta updated if loadStats success?

File: br/pkg/restore/client.go
Comment (3pointer) [inline br/pkg/restore/client.go:1913]: maybe make the count more accurate

**Review by 3pointer — APPROVED** [review]
LGTM

**Comment by hi-rustin** [comment]
I will take a look tomorrow.

**Review by hi-rustin — APPROVED** [review]
The code change looks good to me. But could you please do a real test and post the test result here? Thanks! 👍

And I'd like another set of eyes on it.
/cc @winoros 

/hold

**Pull request merged**
